src/query/storage/m3/consolidators/id_dedupe_map.go

```diff
@@ -129,10 +136,12 @@ func (m *idDedupeMap) doUpdate(
 	}
 	if existsBetter {
 		// Existing result is already better
+		iter.Close()
 		return nil
 	}
 
 	// Override
+	existing.iter.Close()
```

Comment:
There was a `existing.iter.Close` on override in `tagDedupeMap`, I believe the same `Close` was missing here.